metrics/qos/evm/response.go

```diff
@@ -0,0 +1,40 @@
+package evm
+
+import (
+	"github.com/buildwithgrove/path/observation/qos"
+)
+
+// response defines methods needed for response metrics collection.
+// Abstracts proto-specific details from metrics logic.
+// Note: Update when adding new metric requirements.
+type response interface {
+	GetValid() bool
+	GetInvalidReason() qos.EVMResponseInvalidReason
+}
+
+// getResponseFromObservation extracts the response data from an endpoint observation.
+// Returns nil if no response is present.
+// Note: Update when adding new response types.
+func extractEndpointResponseFromObservation(observation *qos.EVMEndpointObservation) response {
+	if observation == nil {
+		return nil
+	}
+
+	if response := observation.GetChainIdResponse(); response != nil {
```

Comment:
Can you do this for every if? I know it's obvious from the code, but much easier to read.


```suggestion
	// handle chain_id response 
	if response := observation.GetChainIdResponse(); response != nil {
```